scripts/pinVersions.js

```diff
@@ -3,35 +3,44 @@ const path = require("path")
 
 const MONOREPO_ROOT = "packages"
 
-const packages = fs.readdirSync(MONOREPO_ROOT)
+const packages = getPackages()
+
+function getPackages() {
+  if (fs.existsSync(MONOREPO_ROOT)) {
+    return fs.readdirSync(MONOREPO_ROOT).map(pkg => path.join(MONOREPO_ROOT, pkg))
+  } else {
+    return ["./"]
+  }
+}
 
 function pinDeps(dependencies) {
-	for (let dependency in dependencies) {
-		if (dependency.startsWith("@budibase")) {
-			dependencies[dependency] = dependencies[dependency].replace("^", "")
-		}
-	}
+  for (let dependency in dependencies) {
+    if (dependency.startsWith("@budibase")) {
```

Comment:
I wonder if we pinned the bb dependencies in our package file, would lerna respect that when it does a release and also use a pinned version. If so it could do away with the need for pinning these altogether

Reply:
I wondered that, I think pinning as part of the build is a good start here, ideally I think it'd be good if we just didn't need to do this stuff, as per how we discussed, building externally and then passing the build + Budibase dependencies into the image and only fetching the non-Budibase dependencies, that way pinning isn't an issue anymore.